satpy/multiscene.py

```diff
@@ -69,8 +80,86 @@ def stack(datasets, weights=None, combine_times=True):
     return base
 
 
-def _stack_weighted(datasets, weights, combine_times):
-    """Stack datasets using weights."""
+def _stack_blended_bands(datasets, weights, combine_times):
+    """Stack datasets with bands blending overlap using weights."""
+    weights = set_weights_to_zero_where_invalid_red(datasets, weights)
+
+    dims = datasets[0].dims
+    attrs = combine_metadata(*[x.attrs for x in datasets])
+
+    if combine_times:
+        if 'start_time' in attrs and 'end_time' in attrs:
+            attrs['start_time'], attrs['end_time'] = _get_combined_start_end_times(*[x.attrs for x in datasets])
+
+    total = weights[0].copy() + 1.e-9
+    for n in range(1, len(weights)):
+        total += weights[n]
+
+    datasets0 = []
+    for n in range(0, len(datasets)):
+        weights[n] /= total
+        datasets0.append(datasets[n].fillna(0))
+        datasets0[n][0] *= weights[n]
+        datasets0[n][1] *= weights[n]
+        datasets0[n][2] *= weights[n]
```

Comment:
```suggestion
    for weight, dataset in zip(weights, datasets):
        weight /= total
        datasets0.append(dataset * weight)
```
I think the array broadcasting mechanism should take care of the weight multiplication. If that works, it would probably mean that we can merge the single and mulitple band cases into one function, right?

Regarding the `fillna` call, I think we should be careful here about what we use as a fill value. In particular, it's probably good to check if the dataset as a defined `_FillValue` attribute already.